frontend/public/components/secrets/create-secret.tsx

```diff
@@ -877,22 +877,42 @@ const secretFormFactory = (secretType: SecretTypeAbstraction) => {
   }
 };
 
-const SecretLoadingWrapper = props => {
-  if (!props.obj.loaded) {
-    return <LoadingBox />;
-  }
-  const secretTypeAbstraction = toTypeAbstraction(props.obj.data);
-  const SecretFormComponent = secretFormFactory(secretTypeAbstraction);
-  const fixed = _.reduce(props.fixedKeys, (acc, k) => ({...acc, k: _.get(props.obj.data, k)}), {});
-  return <StatusBox {...props.obj}>
-    <SecretFormComponent {...props}
-      secretTypeAbstraction={secretTypeAbstraction}
-      obj={props.obj.data}
-      fixed={fixed}
-      explanation={secretFormExplanation[secretTypeAbstraction]}
-    />
-  </StatusBox>;
-};
+class SecretLoadingWrapper extends React.Component<SecretLoadingWrapperProps, SecretLoadingWrapperState> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      formComponent: null,
+      secretType: '',
+    };
+  }
+  componentDidUpdate(prevProps) {
+    // Set the proper secret form component, once the secret is received by Firehose.
+    // 'formComponent' needs to be set only once, to avoid component mounting/unmounting.
+    if (_.isEmpty(prevProps.obj.data) && !_.isEmpty(this.props.obj.data)) {
```

Comment:
Better to check if the form component is not set already instead.

```suggestion
    if (!this.state.formComponent && !_.isEmpty(this.props.obj.data)) {
```